manifests/prepare.pp

```diff
@@ -51,14 +53,87 @@
     }
   }
 
-  # TODO: manage server.cfg contents
-  if $::mcollective_configured {
-    file { '/etc/puppetlabs/mcollective':
-      ensure => directory,
+  # manage client.cfg and server.cfg contents
+  file { '/etc/puppetlabs/mcollective':
+    ensure => directory,
+  }
+  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
+  if $::mco_server_config {
+    $mco_server = '/etc/puppetlabs/mcollective/server.cfg'
+    file { $mco_server:
+      ensure  => file,
+      source  => $::mco_server_config,
+    }
+
+    if $::mco_server_settings {
+      $mco_server_libdir = $::mco_server_settings['libdir']
+      if $mco_server_libdir {
+        ini_setting { 'server/libdir':
+          section => '',
+          setting => 'libdir',
+          path    => $mco_server,
+          value   => "${mco_server_libdir}:/opt/puppetlabs/mcollective/plugins",
```

Comment:
The feedback is great, I'll look into it more tomorrow.

The file resource will ensure it syncs to the original, then updates. I can't think of a scenario where repeatedly running this would result in extra paths.

Michael

> On May 21, 2015, at 17:35, Richard Clamp <notifications@github.com> wrote:
> 
> In manifests/prepare.pp:
> 
> > +  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
> > +  if $::mco_server_config {
> > +    $mco_server = '/etc/puppetlabs/mcollective/server.cfg'
> > +    file { $mco_server:
> > +      ensure  => file,
> > +      source  => $::mco_server_config,
> > +    }
> > +
> > +    if $::mco_server_settings {
> > +      $mco_server_libdir = $::mco_server_settings['libdir']
> > +      if $mco_server_libdir {
> > +        ini_setting { 'server/libdir':
> > +          section => '',
> > +          setting => 'libdir',
> > +          path    => $mco_server,
> > +          value   => "${mco_server_libdir}:/opt/puppetlabs/mcollective/plugins",
> Forgive the driveby, and I may be looking too narrowly, but won't this repeatedly extend libdir? On the first run it'll grab a libdir from a server.cfg, and write this extended value, then on the next run it'll grab the extended value and extend it again, and then on the next run the fact will grab it again, and then extend it again, etc.
> 
> Also libdir is allowed to be specified multiple times in a mcollective configuration file, as the server.cfg isn't really an ini file. This is legit for a server.cfg: https://github.com/puppetlabs/marionette-collective/blob/master/ext/aio/common/server.cfg.dist#L4-L8
> 
> —
> Reply to this email directly or view it on GitHub.
>